script.js

```diff
@@ -1,4 +1,100 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+let numOfDice = 0;
+let gameMode = "waiting for number of players";
+let playerTurn = 0;
+let arrayOfPlayers = [];
+
+//store Player's object to be used for sorting diceRolls and leaderboard
+const storePlayerObjects = (userInput) => {
+  for (let i = 0; i < userInput; i += 1) {
+    arrayOfPlayers.push({ id: i + 1, diceRolls: [] });
+  }
+};
```

Comment:
Great decomposition here Amelia!
It is always good to have your objects named first before pushing, so when we come back to it again we can understand quickly what we were trying to do:

```
var playerDice = { id: i + 1, diceRolls: [] }
arrayOfPlayers.push(playerDice);
```